### DISCUSSION TITLE
Low quality Output

### DISCUSSION THREAD

**Discussion by samuelkurt** [bmaltais/kohya_ss#2587] [Q&A]
I'm training LoRA with standard settings, and the outputs are very low quality. Do i need to change some settings, or do I need some new pictures?

pictures: 50 (512X512, .png)
Network Rank: 200
Everything else is on Standard setting.
Outputs:
![00000-1981233075](https://github.com/bmaltais/kohya_ss/assets/104158296/9d6a9c8c-e8e8-494b-b1c3-876ee3a891ca)
![00001-556890659](https://github.com/bmaltais/kohya_ss/assets/104158296/0053dca7-adda-431e-b759-1a485b3727e4)
![00002-2064220091](https://github.com/bmaltais/kohya_ss/assets/104158296/d6561995-f214-4738-a511-38b3d34de818)

**Comment by b-fission** [comment]
For generating images, there are a few different things you can try. I'll assume you're using automatic1111 webui and your lora is named "last"
- try a lower strength on the lora, example: 60% strength on your lora is `<lora:last:0.6>`
- try some of the past epochs such as `last-000004` instead of the final `last`
- try using negative terms like `(worst quality)` or `(low quality)`
- try hi-res fix with a denoise strength of about 0.3 to 0.4
- try a different sampling method (Euler a, Restart, UniPC)

Some questions about your dataset, if you don't mind.
1. Are the source images considered high-quality? That being: decent resolution (originally 512x512 or better) with no obvious compression artifacts / scaling artifacts / visual anomalies? (those outputs are reminiscent of poorly upscaled jpeg thumbnails)
2. Are you using captions with your source images? If yes, are they manually captioned or auto captioned?
3. What is the repeat count on your image folder?
4. Do the colors of the outputs resemble the source images?

**Reply by samuelkurt** [reply]
Tryed your solutions, nothing worked.

Here are your answers:
1) They've been all downscaled to 512X512 (Ones opend, the look very pixeld)
2) No, I don't
3) 40
4) Sometimes

Do all pictures have to have the same Resolution for a good Training, or can the be Different?

**Reply by b-fission** [reply]
> 1. They've been all downscaled to 512X512 (Ones opend, the look very pixeld)

Did they look pixelated at their original size? What tool did you use to downscale them? I'd probably try another tool if the downscaled images look bad because it will affect the trained result.

> 2. No, I don't

Then what are the name and class words for your image folder?

> 3. 40

40 repeats seems high for 50 images, and would generate only 1 epoch if you're using the default max step count of 1600. This might also indicate overtraining, and did you have no past epochs to test? ... I'd suggest something lower like 5 repeats. Also configure kohya to do a sample image (or a few) after each epoch so you can get a better idea on the quality of the lora while it is training.
 
> 4. Sometimes

Okay.

> Do all pictures have to have the same Resolution for a good Training, or can the be Different?

They can be different, but kohya will automatically resize them, constraining to an overall pixel count (the "max resolution" setting which is defaulted at 512x512). You can also use non-square images like portrait or landscape, and kohya can train on those to generate non-square images.

**Comment by samuelkurt** [comment]
1) No, they weren’t
2) Name:qwertzu class:robot

**Comment by PetrHuu** [comment]
Did you use batch size 1 in training?

**Reply by samuelkurt** [reply]
Yes, I did.
